lib/rules/declaration-block-trailing-semicolon/index.js

```diff
@@ -49,6 +49,12 @@ const rule = function(expectation) {
         if (node.parent.raws.semicolon) {
           return;
         }
+
+        // auto-fix
+        if (context.fix) {
+          node.parent.raws.semicolon = ";";
```

Comment:
I believe `raws.semicolon` property expects a boolean.

You need to account for the `"never"` option too i.e. the trailing semicolon is removed when fixed (as this is a valid choice). This will be revealed when you add `fix: true` to the two existing rejects.